pkg/generated/openapi/zz_generated.openapi.go

```diff
@@ -3887,7 +3887,7 @@ func schema_sigsk8sio_gateway_api_apis_v1_GatewayClassStatus(ref common.Referenc
 							},
 						},
 						SchemaProps: spec.SchemaProps{
-							Description: "Conditions is the current status from the controller for this GatewayClass.\n\nControllers should prefer to publish conditions using values of GatewayClassConditionType for the type of each Condition.",
+							Description: "Conditions is the current status from the controller for this GatewayClass.\n\nControllers should prefer to publish conditions using values of GatewayClassConditionType for the type of each Condition.\n\n<gateway:util:excludeFromCRD> Notes for implementors:\n\nConditions are a listType `map`, which means that they function like a map with a key of the `type` field _in the k8s apiserver_.\n\nThis means that implementations must obey some rules when updating this section.\n\n* Implementations MUST perform a read-modify-write cycle on this field\n  before modifying it. That is, when modifying this field, implementations\n  must be confident they have fetched the most recent version of this field,\n  and ensure that changes they make are on that recent version.\n* Implementations MUST NOT remove or reorder Conditions that they are not\n  directly responsible for. For example, if an implementation sees a Condition\n  with type `special.io/SomeField`, it MUST NOT remove, change or update that\n  Condition.\n* Implementations MUST always _merge_ changes into Conditions of the same Type,\n  rather than creating more than one Condition of the same Type.\n* Implementations MUST always update the `observedGeneration` field of the\n  Condition to the `metadata.generation` of the Gateway at the time of update creation.\n* If the `observedGeneration` of a Condition is _greater than_ the value the\n  implementation knows about, then it MUST NOT perform the update on that Condition,\n  but must wait for a future reconciliation and status update. (The assumption is that\n  the implementation's copy of the object is stale and an update will be re-triggered\n  if relevant.)\n\n</gateway:util:excludeFromCRD>",
```

Comment:
hum, fun. I though that once we excluded from the CRD, it would be also excluded from the openapi definition.

Am I missing something?

Reply:
Sadly not, we only have custom CRD gen right now - does not effect OpenAPI gen: https://github.com/kubernetes-sigs/gateway-api/blob/8fe8316f5792a7830a49c800f89fe689e0df042e/pkg/generator/main.go#L256-L259. This would be ripe for some cleanup though if anyone has time.

Reply:
yeah, I was wondering if do we really care about oapi generation, as FWIW the CRD is what is exposed for users, and IIUC the oapi generation is just used as part of the apply configuration (as far as I can see it)

If not, from my side I am fine with Nick's change as they are excluded from the CRD, and I have an issue already to start working on the CRD field meta GEP

Reply:
Yes, I don't think we've paid much attention to OpenAPI generation in the past; that's why we don't have custom generation for that part.